RecoDataProducts/inc/STMWaveform.hh

```diff
@@ -0,0 +1,51 @@
+#ifndef RecoDataProducts_STMWaveform_hh
+#define RecoDataProducts_STMWaveform_hh
+//
+// Data product that represents the digitized waveforms coming from the STM detectors
+// This is used for both unsuppressed and zero-suppressed waveforms
+//
+// We will assign the channel number to the STMWaveformCollection since we
+// plan to keep the waveforms from each detector separate
+//
+
+// C++ includes
+#include <iostream>
+#include <vector>
+#include <array>
+#include <Rtypes.h>
+
+#include "Offline/DataProducts/inc/STMChannel.hh"
+
+namespace mu2e {
+
+  class STMWaveform {
+  public:
+    STMWaveform() : _trigTimeOffset(0), _adcs(std::vector<int16_t>()){};
+
+    STMWaveform(uint32_t trigTimeOffset, std::vector<int16_t> adcs) : _trigTimeOffset(trigTimeOffset), _adcs(adcs) {};
+
+
+    uint32_t trigTimeOffset() const { return _trigTimeOffset; }
+    const std::vector<int16_t>& adcs() const { return _adcs; }
+
+  private:
+    uint32_t _trigTimeOffset; // time offset from EWT? to first ADC value [ct]
+    std::vector<int16_t> _adcs; // vector of ADC values for the waveform
+  };
+
+  class STMWaveformCollection {
```

Comment:
since STMWaveformCollection is its own class (not just a typedef) I recommend splitting to its own header file.  Also, do you ever forsee a data product that's a collection (std::vector) of STMWaveformCollection?  That could result in some awkward names.

Reply:
I'm happy to move STMWaveformCollection into its own header file. I don't ever forsee us making a collection of STMWaveformCollections. I'd like to keep things fixed so that we have one STMWaveformCollection for HPGe and another for LaBr. Each will be treated slightly differently so I don't see a reason we would want to loop through a collection of collections.